### DISCUSSION TITLE
Error when getting IngressClass nginx: ingressclasses.networking.k8s.io "nginx" not found in nginx-ingress deployments

### DISCUSSION THREAD

**Discussion by iamraj007** [nginx/kubernetes-ingress#2030] [Q&A]
Was trying this but was stuck after deploying nginx-ingress.yaml with below error, is this sorted issue or am i doing it wrong? 
https://github.com/nginxinc/kubernetes-ingress/tree/master/deployments

kubectl create -f deployments/deployment/nginx-ingress.yaml

 kubernetes-ingress]$  kubectl logs pod/nginx-ingress-ddd64b786-8bdfz -n nginx-ingress
I0930 14:35:14.340272       1 main.go:271] Starting NGINX Ingress controller Version=1.12.1 GitCommit=6f72db6030daa9afd567fd7faf9d5fffac9c7c8f Date=2021-09-08T13:39:53Z PlusFlag=false
W0930 14:35:14.370136       1 main.go:310] The '-use-ingress-class-only' flag will be deprecated and has no effect on versions of kubernetes >= 1.18.0. Processing ONLY resources that have the 'ingressClassName' field in Ingress equal to the class.
W0930 14:35:14.379336       1 warnings.go:70] networking.k8s.io/v1beta1 IngressClass is deprecated in v1.19+, unavailable in v1.22+; use networking.k8s.io/v1 IngressClassList
F0930 14:35:14.379783       1 main.go:314] **Error when getting IngressClass nginx: ingressclasses.networking.k8s.io "nginx" not found**

 kubernetes-ingress]$  kubectl get all -n nginx-ingress
NAME                                READY   STATUS             RESTARTS   AGE
pod/nginx-ingress-ddd64b786-8bdfz   0/1     CrashLoopBackOff   8          16m

NAME                            READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/nginx-ingress   0/1     1            0           16m

NAME                                      DESIRED   CURRENT   READY   AGE
replicaset.apps/nginx-ingress-ddd64b786   1         1         0       16m

kubernetes-ingress]$  kubectl describe pod/nginx-ingress-ddd64b786-8bdfz -n nginx-ingress
Name:         nginx-ingress-ddd64b786-8bdfz
Namespace:    nginx-ingress
Priority:     0
Node:         ip-172-XXXXXX3.us-west-2.compute.internal/172.xxxxx13
Start Time:   Thu, 30 Sep 2021 14:24:22 +0000
Labels:       app=nginx-ingress
              pod-template-hash=ddd64b786
Annotations:  kubernetes.io/psp: eks.privileged
Status:       Running

Controlled By:  ReplicaSet/nginx-ingress-ddd64b786
Containers:
  nginx-ingress:
    Container ID:  docker://8b954ba0a1609321774d34ce3db625153f3d8317bb4a1d24e417c003e1089992
    Image:         nginx/nginx-ingress:1.12.1
    Image ID:      docker-pullable://nginx/nginx-ingress@sha256:9e922a06363643d252978d7646ccb4c031c0901b0d59bc2eba27c3d7072e3882
    Ports:         80/TCP, 443/TCP, 8081/TCP, 9113/TCP
    Host Ports:    0/TCP, 0/TCP, 0/TCP, 0/TCP
    Args:
      -nginx-configmaps=$(POD_NAMESPACE)/nginx-config
      -default-server-tls-secret=$(POD_NAMESPACE)/default-server-secret
    State:          Waiting
      Reason:       **CrashLoopBackOff**
    Last State:     Terminated
      Reason:       Error
      Exit Code:    255

    Ready:          False
    Restart Count:  8
    Readiness:      http-get http://:readiness-port/nginx-ready delay=0s timeout=1s period=1s #success=1 #failure=3
    Environment:
      POD_NAMESPACE:  nginx-ingress (v1:metadata.namespace)
      POD_NAME:       nginx-ingress-ddd64b786-8bdfz (v1:metadata.name)
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-tdzv6 (ro)
Conditions:
  Type              Status
  Initialized       True
  Ready             False
  ContainersReady   False
  PodScheduled      True
Volumes:
  kube-api-access-tdzv6:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   BestEffort
Node-Selectors:              <none>
Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                             node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type     Reason     Age                From               Message
  ----     ------     ----               ----               -------
  Normal   Scheduled  18m                default-scheduler  Successfully assigned nginx-ingress/nginx-ingress-ddd64b786-8bdfz to ip-172-16-4-113.us-west-2.compute.internal
  Normal   Started    17m (x4 over 18m)  kubelet            Started container nginx-ingress
  Normal   Pulled     16m (x5 over 18m)  kubelet            Container image "nginx/nginx-ingress:1.12.1" already present on machine
  Normal   Created    16m (x5 over 18m)  kubelet            Created container nginx-ingress
  Warning  BackOff    3m (x78 over 18m)  kubelet            Back-off restarting failed container

**Comment by jasonwilliams14** [comment]
@iamraj007 Hello there
did you install our `ingressClass`?
You can run:
```
kubectl get ingressclass
```

Can you show us what that output reveals?
Thank you.

**Comment by jjngx** [comment]
No answer in 4 years. Closing.
